src/hooks/use-arrow-key-navigation.ts

```diff
@@ -215,10 +215,11 @@ export function useArrowKeyNavigation(
         lastFocusedItem.current.focus();
         return;
       }
+
       const elements = getNavigableElements();
       const targetIndex = elements.indexOf(event.target as HTMLElement);
       if (targetIndex >= 0) {
-        updateTabIndexes(elements, targetIndex);
+        updateTabIndexes(elements, targetIndex, autofocus);
```

Comment:
This is what was causing the options to not me set as `lastFocusedItem` when being clicked instead of selected with the keyboard.